### DISCUSSION TITLE
antv graph，dagre layout超时问题

### DISCUSSION THREAD

**Discussion by owensulei3841** [antvis/G6#3529] [Q&A]
请问，我现在自定义了一些节点卡片，长400px, 高400px, 我的数据有1700个节点，7000个边，在rendering图的时候，特别的慢，然后chrome会超时，会弹出一个page unresponsive的弹出框，请问为什么会这样？是我的自定义节点太复杂太大？有什么好的办法fix这问题嘛？？谢谢

layout: {
      type: 'dagre',
      rankdir: 'LR',
      align: 'DL',
      // workerEnabled: true,
      nodesepFunc: (cfg: any) => {
        return 200;
      },
      ranksepFunc: () => 200,
    },

**Comment by cliffordfajardo** [comment]
### English translation:
> Excuse me, I have customized some node cards now, with a length of 400px and a height of 400px. My data has 1700 nodes and 7000 edges. When rendering the graph, it is very slow, and then chrome will timeout and a page unresponsive will pop up. A pop-up box, why is this happening? Is my custom node too complex and too big? Is there any good way to fix this problem? ? thanks


@owensulei3841 -  would you be able to share a screenshot image, GIF or mp4 video of the interaction?
This Web browser extension allows you to record or take screenshot
- https://chrome.google.com/webstore/detail/screencastify-screen-vide/mmeijimgabbpbgpdklnllpncmdofkcpn

I'm curious to know more about your custom node 🤔 

### Question(s)
1. Do you have any optimization settings enabled?
   - for example: `enabledOptimize`?
<img width="926" alt="CleanShot 2022-03-01 at 14 29 49@2x" src="https://user-images.githubusercontent.com/6743796/156259824-a55e3ea9-d4ce-4d53-a255-c6600ecd2f47.png">

2. Have you tried any of these optimization tips?
- https://g6.antv.vision/en/docs/manual/middle/elements/nodes/custom-node/#optimize-the-performance
```js
//
  modes: {
    default: [
      {
        type: "zoom-canvas",
        enableOptimize: true,
        optimizeZoom: 0.9
      },
      {
        type: "drag-canvas",
        enableOptimize: true
      },
      "drag-node",
      "brush-select"
    ]
  }
```



3. Can you share a small codesanbox snippet of example of your custom node?
- https://codesandbox.io/s
